blocks/search/search.js

```diff
@@ -104,172 +100,176 @@ function addSearchByListeners(wrapper, form) {
+    const type = searchType === 'cross' ? 'cross-reference' : 'parts';
+    const titleText = `${titleContent} ${searchType === 'cross' ? `${type}: "${query}"` : `${make || ''} ${model || ''} ${query} ${type}`}`;
+    titleElement.textContent = titleText;
+  }
+};
+
+const handleShowMoreButtons = (resultCount, targetOffset, resultsSection) => {
+  const buttonTextContent = getTextLabel('pagination_button');
+  const currentAmount = document.querySelectorAll('.product-card').length;
+  const displayedTextContent = getTextLabel('pagination_text').replace('[$]', currentAmount);
+
+  const resultsCountElement = document.querySelector('.pagination .top-results-text');
+  if (resultsCountElement) {
+    resultsCountElement.innerText = displayedTextContent;
+  }
+
+  if (targetOffset === 0) {
+    const bottomMoreBtn = createElement('button', { classes: ['more-button', 'bottom-more-button'] });
+    bottomMoreBtn.textContent = buttonTextContent;
+    resultsSection?.appendChild(bottomMoreBtn);
+    bottomMoreBtn.onclick = () => getAndApplySearchResults({ isFirstSet: false });
+  }
+
+  if (resultCount < parseInt(SEARCH_CONFIG.MAX_PRODUCTS_PER_QUERY)) {
+    document.querySelectorAll('.more-button').forEach((moreBtn) => moreBtn.remove());
+  }
+};
+
+const displayNoResults = (titleElement, searchResultsSection, filtersWrapper, query) => {
+  if (titleElement) {
+    const titleText = getTextLabel('no_results_title').replace('[$]', `${query}`);
+    titleElement.innerText = titleText;
+  }
+
+  const fragment = document.createRange().createContextualFragment(noResultsTemplate);
+  searchResultsSection?.classList.add('no-results');
+  if (filtersWrapper && searchResultsSection) {
+    searchResultsSection.insertBefore(fragment, filtersWrapper);
+  }
+};
+
+const updateFilters = (filtersWrapper, categories) => {
+  const filters = buildFilter(categories);
+  if (filters) {
+    filtersWrapper.innerHTML = '';
+    filtersWrapper.append(filters);
+  }
+};
+
 async function getAndApplyFiltersData(form) {
   const makeSelect = form.querySelector(`.${blockName}__make-filter__select`);
   const modelsSelect = form.querySelector(`.${blockName}__model-filter__select`);
-  const makeItems = [];
-  const filters = await getJsonFromUrl(FILTERS_DATA);
-  const { data } = filters || {};
-  if (!data) return;
-  data.forEach((item) => {
-    const itemModels = item.Models !== 'null' ? item.Models.split(',') : [];
-    modelsItems.push({ Make: item.Make, Models: itemModels });
-    makeItems.push(item.Make);
-  });
+  const fetchedMakeFacets = await fetchFilterFacets({ field: 'MAKE' });
+  const makeFacets = fetchedMakeFacets.facets.facets;
+  const makeItems = makeFacets.map((facet) => facet.key);
   populateFilter(makeSelect, makeItems);
-  noOthersItems = makeItems.filter((item) => item !== 'Others');
-  makeSelect.onchange = (e) => {
+  makeSelect.onchange = async (e) => {
     const isNotNull = e.target.value !== 'null';
```

Comment:
I know this is not on you, but if you do another commit please change this:
`const isNotNull = e.target.value !== 'null';`
`if (!isNotNull) {`
To:
`const isNull = e.target.value === 'null';`
`if (isNull) {`

Reply:
Done